template/src/Autovis/Controls.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import './controls.css';
+import toPercent from './toPercent';
+import printTime from './printTime';
+
+const getEventXPositionRatio = (e) => {
+  const rect = e.currentTarget.getBoundingClientRect();
+  const mouseXRel = e.clientX - rect.x;
+  const positionRatio = Math.min(Math.max(0, mouseXRel / rect.width), 1);
+  return positionRatio;
+};
+
+export default ({ onPlay, onPause, paused, onSeek, currentTime, duration }) => (
+  <div className="controls">
+    <button
+      className="play"
+      onClick={() => {
+        paused ? onPlay() : onPause();
+      }}
+    >
+      {paused ? "Play" : "Pause"}
```

Comment:
quotes (double quote only directly in property shorthand like className="play", everywhere else in JSX should be single quote, including stuff like className={thing1 + ' asdf'})

Reply:
kk woops.